src/collection.js

```diff
@@ -87,6 +87,7 @@ module.exports = function(options) {
               repo.checkout(newbranch, function() {
                 repo.push(" origin " + newbranch, function() {
                   console.log("Branch created & pushed to origin")
+                  branch = newbranch
```

Comment:
Is this the desired behaviour? This would be changing the underlying collection (altering future loads and saves to that original collection), whereas in the tests/branch.js, we load the newly created branch explicitly, like so - 

`framework.collection({name: 'test'}) `
`.then(c => framework.createBranch(c, 'dev'))`
`.then(c => framework.collection({name: 'test', branch: 'dev'}))`

This stops any side effects from using the original 'master' collection which would now have changed branch, which might be an annoyance to keep track of when we play around with lots of different branches.

`framework.collection({name: 'test', branch: 'master'}) `
`.then(c => framework.createBranch(c, 'dev'))`

With proposed changes, at this point, the 'master' collection created follows the new branch implicitly. Obviously the semantics are up for interpretation at this point but seems like it should be explicit?